datatable/frame.py

```diff
@@ -165,23 +141,23 @@ def _fill_from_source(self, src, names, stypes):
             self._fill_from_list(list(src.values()), names=tuple(src.keys()),
                                  stypes=stypes)
         elif isinstance(src, core.DataTable):
-            src._set_names(names)
             self._dt = src
+            self.names = names
         elif isinstance(src, str):
             srcdt = datatable.fread(src)
             if names is None:
                 names = srcdt.names
             self._dt = srcdt.internal
-            self._dt._set_names(names)
+            self.names = names
```

Comment:
So for the `aggregator` I should also use `self.names = names` rather than the `_set_names()` or any other options?

Reply:
`_set_names()` doesn't really exist anymore, so yes, use the `names` setter